client/parties/action-creators.ts

```diff
@@ -110,3 +110,23 @@ export function acceptPartyInvite(partyId: string): ThunkAction {
     })
   }
 }
+
+export function leaveParty(): ThunkAction {
+  return dispatch => {
+    dispatch({
+      type: '@parties/leavePartyBegin',
```

Comment:
should probably include info about what party is being left

Reply:
Initially I had `partyId` here, but later removed it since I realized I don't use it anywhere. And I don't mean that alone makes it useless, but once I realized I don't even send the `partyId` to the server, I stopped looking at this as a "leave _this_ party" action, and more like a "leave the client's party, whatever it is, and if there's one".

Now I'm not sure if that's the best way of looking at this action, but that's the least amount of information I needed to perform it. The main reason why I'm skeptical of approaching it like I did is when looking at the HTTP route. Currently I defined the API route to leave the party as `DELETE /api/1/parties`. While speaking from a strictly REST perspective, `DELETE /api/1/parties/:partyId/:clientId` maybe makes the most amount of sense to me?

Reply:
Yes, I think that URL makes more sense (I'd view `:clientId` as pretty optional, but I don't think it hurts to include it)

Reply:
I think in general you should try to include information about what entity the client is trying to update, rather than figuring this out all automatically on the server, simply because it gives us a better opportunity of identifying bugs and mismatched client state. It also prevents a class of timing issues, because the server will only update the state about the thing the client was explicit it was talking about

Reply:
Yeah, I only included `:clientId` because without it, the URL seemed like a party itself was being deleted :d